npm/test/folderinforetrieval.test.ts

```diff
@@ -67,7 +89,7 @@ test("folder retriever updates the page with correct folder information if there
   triggerInputEvent(folderRetriever!)
 
   const folderSummaryElement: HTMLElement | null = document.querySelector(
-    ".govuk-summary-list__row"
+    ".drag-and-drop__success"
```

Comment:
We should add some additional unit test to make sure the page state is correct when:

- a user drags and drops an empty folder: `test("folder retriever updates the page with correct folder information if there are no files", () => { ... etc }`
- a user drags and drops a non-folder object:  `test("folder retriever updates the page with correct folder information if there are is a non file object", () => { ... etc }`

Not sure such test are possible, but they should be included as unit tests as they form part of the logic for the behaviour.

Reply:
Yes. I was trying to remember why I left out the failure test, but now I do. It was because when there is an error in the actual code, it throws an exception which doesn't allow the test to complete; I tried a few things but could not get past the exception. This was something that Suzanne said she was going to have have a think about but we should leave it for now.

Reply:
OK.